CI: `TestPlotScatter_Gl.testGridVisualization` repeatedly fails for py312/pyside6 on ubuntu

Run on current `main`: https://github.com/silx-kit/silx/actions/runs/29915893582/job/88909545651

```
_ TestPlotScatter_Gl.testGridVisualization

[...]

 UBFAILED(name='grid fast x, +x -y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - ValueError: Error calling Python override of QOpenGLWidget::paintGL(): cannot reshape array of size 8512 into shape (19,111,4)
SUBFAILED(name='grid fast x, -x -y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: array([2]) != (0,)
SUBFAILED(name='grid fast x, -x +y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: array([5]) != (0,)
SUBFAILED(name='grid fast y, +x +y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: array([1]) != (0,)
SUBFAILED(name='grid fast y, +x -y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: unexpectedly None
SUBFAILED(name='grid fast y, -x -y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: unexpectedly None
SUBFAILED(name='grid fast y, -x +y', visualization='REGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: array([4]) != (0,)
SUBFAILED(name='horizontal line', visualization='IRREGULAR_GRID') ../../opt/hostedtoolcache/Python/3.12.13/x64/lib/python3.12/site-packages/silx/gui/plot/test/test_plotwidget.py::TestPlotScatter_Gl::testGridVisualization - AssertionError: Tuples differ: (array([0]),) != (1,)
```

Maybe (?) related to matplotlib release v3.11.1 from 18th of July 2026.

--- a/src/silx/gui/utils/matplotlib.py
+++ b/src/silx/gui/utils/matplotlib.py
@@ -35,7 +35,6 @@ __license__ = "MIT"
 __date__ = "02/05/2018"
 
 
-import io
 import matplotlib
 import numpy
 
@@ -54,6 +53,7 @@ from matplotlib.font_manager import FontProperties
 from matplotlib.mathtext import MathTextParser
 from matplotlib.ticker import ScalarFormatter as _ScalarFormatter
 from matplotlib import figure, font_manager
+from matplotlib.backends.backend_agg import FigureCanvasAgg
 
 
 class DefaultTickFormatter(_ScalarFormatter):
@@ -147,13 +147,12 @@ def rasterMathText(
         verticalalignment="top",
     )
     text.set_linespacing(linespacing)
-    with io.BytesIO() as buffer:
-        fig.savefig(buffer, dpi=dotsPerInch, format="raw")
-        canvas_width, canvas_height = fig.get_window_extent().max
-        buffer.seek(0)
-        image = numpy.frombuffer(buffer.read(), dtype=numpy.uint8).reshape(
-            int(canvas_height), int(canvas_width), 4
-        )
+
+    canvas = FigureCanvasAgg(fig)
+    buffer, (canvas_width, canvas_height) = canvas.print_to_buffer()
+    image = numpy.frombuffer(buffer, dtype=numpy.uint8).reshape(
+        canvas_height, canvas_width, 4
+    )
 
     # RGB to inverted R channel
     array = 255 - image[:, :, 0]

